packages/xo-web/src/xo-app/menu/index.js

```diff
@@ -66,10 +71,25 @@ export default class Menu extends Component {
       window.removeEventListener('resize', updateCollapsed)
       this._removeListener = noop
     }
+
+    this._subscribeMissingPatches()
   }
 
   componentWillUnmount() {
     this._removeListener()
+    this._unsubscribeMissingPatches()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      !isEqual(
+        map(prevProps.hosts, 'id').sort(),
+        map(this.props.hosts, 'id').sort()
+      )
+    ) {
+      this.setState({ missingPatches: {} })
```

Comment:
To initialize the state, if the hosts change. 
If we don't initialize it, it will contain the previous state of previous hosts.

Reply:
This can be handled in `_subscribeMissingPatches`.